src/routes/PipelineRun/PipelineRun.tsx

```diff
@@ -13,22 +13,68 @@ import { useDocumentTitle } from "@/hooks/useDocumentTitle";
 import { useBackend } from "@/providers/BackendProvider";
 import { useComponentSpec } from "@/providers/ComponentSpecProvider";
 import { type RunDetailParams, runDetailRoute } from "@/routes/router";
-import { useFetchExecutionInfo } from "@/services/executionService";
+import {
+  useFetchExecutionInfo,
+  useFetchPipelineRun,
+} from "@/services/executionService";
 import { getBackendStatusString } from "@/utils/backend";
 import type { ComponentSpec } from "@/utils/componentSpec";
 
 const PipelineRun = () => {
   const { setComponentSpec, clearComponentSpec, componentSpec } =
     useComponentSpec();
   const { backendUrl, configured, available } = useBackend();
-  const { id: rootExecutionId } = runDetailRoute.useParams() as RunDetailParams;
+  const { id } = runDetailRoute.useParams() as RunDetailParams;
+  const [rootExecutionId, setRootExecutionId] = useState<string>(id);
+  const [triedAsRunId, setTriedAsRunId] = useState<boolean>(false);
 
-  const { data, isLoading, error, refetch } = useFetchExecutionInfo(
+  // First try to fetch as root_execution_id
+  const {
+    data: executionData,
+    isLoading: isExecutionLoading,
+    error: executionError,
+    refetch: refetchExecution,
+  } = useFetchExecutionInfo(
     rootExecutionId,
     backendUrl,
     false,
+    configured,
+    available,
   );
 
+  // If fetching as root_execution_id fails, try as run_id
+  const shouldFetchAsRunId = executionError && !triedAsRunId;
+  const {
+    data: pipelineRunData,
+    isLoading: isPipelineRunLoading,
+    error: pipelineRunError,
+  } = useFetchPipelineRun(
+    shouldFetchAsRunId ? id : undefined,
+    backendUrl,
+    configured,
+    available,
+  );
```

Comment:
Looking at the logic here and in `useFetchPipelineRun` I think it might make more sense to expose `enabled` as a prop and use that to control whether the query executes. So here we'd end up with something like:

```suggestion
  } = useFetchPipelineRun(
    id,
    backendUrl,
    shouldFetchAsRunId && configured && available,
  );
```

Where `useFetchPipelineRun` is something like:

```
export const useFetchPipelineRun = (
  runId: string,
  backendUrl: string,
  enabled: boolean = true
) => {
  return useQuery<PipelineRunResponse>({
    queryKey: ["pipeline-run", runId],
    queryFn: () => fetchPipelineRun(runId, backendUrl),
    enabled: enabled,
    refetchOnWindowFocus: false,
  });
};